sklearn/impute.py

```diff
@@ -39,10 +41,17 @@
 
 def _get_mask(X, value_to_mask):
     """Compute the boolean mask X == missing_values."""
-    if value_to_mask == "NaN" or np.isnan(value_to_mask):
-        return np.isnan(X)
+    if value_to_mask is np.nan:
+        if X.dtype.kind in ("i", "u", "f"):
```

Comment:
This is confusing. You can't have np.nan in an integer array.

Reply:
The alternative is to directly return `np.zeros(X.shape, dtype=bool)` for integer types? (because we still need to handle the case I think)

Reply:
I am +1 with treating integer with a specific case as @jorisvandenbossche . That would improve readability.